tests/libgit2/object/shortid.c

```diff
@@ -49,3 +48,43 @@ void test_object_shortid__select(void)
 
 	git_buf_dispose(&shorty);
 }
+
+void test_object_shortid__core_abbrev(void)
+{
+	git_oid full;
+	git_object *obj;
+	git_buf shorty = {0};
+	git_config *cfg;
+
+	cl_git_pass(git_repository_config(&cfg, _repo));
+	git_oid__fromstr(&full, "ce013625030ba8dba906f756967f9e9ca394464a", GIT_OID_SHA1);
+	cl_git_pass(git_object_lookup(&obj, _repo, &full, GIT_OBJECT_ANY));
+
+	cl_git_pass(git_config_set_string(cfg, "core.abbrev", "auto"));
+	cl_git_pass(git_object_short_id(&shorty, obj));
+	cl_assert_equal_i(7, shorty.size);
+	cl_assert_equal_s("ce01362", shorty.ptr);
+
+	cl_git_pass(git_config_set_string(cfg, "core.abbrev", "off"));
+	cl_git_pass(git_object_short_id(&shorty, obj));
+	cl_assert_equal_i(40, shorty.size);
+	cl_assert_equal_s("ce013625030ba8dba906f756967f9e9ca394464a", shorty.ptr);
+
+	cl_git_pass(git_config_set_string(cfg, "core.abbrev", "4"));
+	cl_git_pass(git_object_short_id(&shorty, obj));
+	cl_assert_equal_i(4, shorty.size);
+	cl_assert_equal_s("ce01", shorty.ptr);
+
+	cl_git_pass(git_config_set_string(cfg, "core.abbrev", "3"));
+	cl_git_fail(git_object_short_id(&shorty, obj));
+	cl_git_pass(git_config_set_string(cfg, "core.abbrev", "41"));
```

Comment:
In fact, git handles very large `core.abbrev` settings by converting them to the max length.

```
git -c core.abbrev=4242 log
commit 1738f891a1eca8d94ba1cffcabd1ec41af122c52 (HEAD -> main, origin/main, origin/HEAD)
Merge: 27d82f15cdf572399fe0c58853e497642096957a 7668c13552cc1090608726bb871f57ac4f7eab83
...
```